### PR TITLE
Extending initialization to support EU realm

### PR DESCRIPTION
**Pull request by fermelone**
Extended the initialization of the dogapi to support other endpoints such as the EU API, given that the dogapi already supports this. https://brettlangdon.github.io/node-dogapi/#dogapi-initialize
Also added the parameter and env options.

### CODE REVIEW

File: lib/reporters.js [hunk 1]
```diff
@@ -21,16 +21,17 @@ NullReporter.prototype.report = function(series, onSuccess, onError) {
 // DataDogReporter
 //
 
-function DataDogReporter(apiKey, appKey, agent) {
+function DataDogReporter(apiKey, appKey, agent, apiHost) {
     apiKey = apiKey || process.env.DATADOG_API_KEY;
     appKey = appKey || process.env.DATADOG_APP_KEY;
+    apiHost = apiHost || process.env.DATADOG_API_HOST;
```
Comment (jpadhye) [inline lib/reporters.js]: Wouldn't `apiHost` need a default value as it's an optional parameter in the init function.

Comment (wouterlemcke) [reply lib/reporters.js:27]: dogapi is already taking care of api_host being undefined. Should be fine

**Comment by alxch-** [comment]
Hi,

Do you have any news about this? It would be great to have this feature, and the PR looks correct!

**Comment by alxch-** [comment]
👋  me again, would it make sense to also update the `BufferedMetricsLogger` constructor [here](https://github.com/fermelone/node-datadog-metrics/blob/master/lib/loggers.js#L41) so that the `apiHost` can be passed as an option when creating the metrics logger?

**Comment by jekusz** [comment]
I also think it would be nice to fix this line in loggers.js:
```
	this.reporter =
		opts.reporter || new DataDogReporter(opts.apiKey, opts.appKey, opts.agent, opts.apiHost);
```

**Comment by Skrohk** [comment]
Up ! No news about this ? It would be useful ! Thanks !

**Comment by ErikBoesen** [comment]
Hi! Taking over as a new maintainer for this project—please resolve the failing Travis build and I will merge this.

**Review by ErikBoesen — APPROVED** [review]

**Comment by rpelliard** [comment]
Hello @ErikBoesen, I think the failing build is unrelated to these changes. I just submitted https://github.com/dbader/node-datadog-metrics/pull/60 to fix the build

**Review by ErikBoesen — APPROVED** [review]

**Comment by rpelliard** [comment]
@ErikBoesen would it be possible to make a new release including these changes ? Thanks !

**Comment by ErikBoesen** [comment]
Sorry for the delay—just published v0.9.0! @rpelliard

**Comment by jektvik** [comment]
Guys, would you want to take another look at this. The PR is merged alright, but the problem is hardly fixed.
The PR only covers DataDogReporter but that class is not even exported as part of the public package api.

**Comment by ErikBoesen** [comment]
@jektvik That class, while not exported through the public package API, is used/called through `BufferedMetricsLogger` and by extension through the package `init` function and elsewhere.

**Comment by jektvik** [comment]
@ErikBoesen well in this PR a support for the `apiHost` parameter is introduced but nowhere in the entire repo is this parameter passed in from the calling code which renders it essentially dead code.
The only way to pass in the other DD host address is by knowing the 'magic' environment variable but who on earth is going to know to look for its existence deep in the bowels of the code-base.

**Comment by ErikBoesen** [comment]
@jektvik Good catch, I see that we never added the parameter [here](https://github.com/dbader/node-datadog-metrics/blob/master/lib/loggers.js#L41), so I've just done so.

**Comment by ErikBoesen** [comment]
I've pushed v0.9.2 to fix this issue.

**Pull request merged**